client/signup/jetpack-connect/plans.jsx

```diff
@@ -60,6 +60,39 @@ const Plans = React.createClass( {
 			( site.plan.product_slug === 'jetpack_business' || site.plan.product_slug === 'jetpack_premium' );
 	},
 
+	selectFreeJetpackPlan() {
+		const selectedSite = this.props.sites.getSelectedSite();
+		this.props.recordTracksEvent( 'calypso_jpc_plans_submit_free', {
+			user: this.props.userId
+		} );
+		if ( isCalypsoStartedConnection( this.props.jetpackConnectSessions, selectedSite.slug ) ) {
+			page.redirect( CALYPSO_REDIRECTION_PAGE + selectedSite.slug );
+		} else {
+			const { queryObject } = this.props.jetpackConnectAuthorize;
+			this.props.goBackToWpAdmin( queryObject.redirect_after_auth );
+		}
+	},
+
+	selectPlan( cartItem ) {
+		const selectedSite = this.props.sites.getSelectedSite();
+		const checkoutPath = `/checkout/${ selectedSite.slug }`;
+		if ( cartItem.product_slug === 'jetpack_free' ) {
+			return this.selectFreeJetpackPlan();
+		}
+		if ( cartItem.product_slug === 'jetpack_premium' ) {
+			this.props.recordTracksEvent( 'calypso_jpc_plans_submit_99', {
+				user: this.props.userId
+			} );
+		}
+		if ( cartItem.product_slug === 'jetpack_business' ) {
+			this.props.recordTracksEvent( 'calypso_jpc_plans_submit_299', {
+				user: this.props.userId
+			} );
+		}
+		upgradesActions.addItem( cartItem );
```

Comment:
Is this necessary? Couldn't we just redirect to `page( `/checkout/${ selectedSiteSlug }/${ getPlanPath( plan ) || '' }` );` as done here: https://github.com/Automattic/wp-calypso/blob/533265f38693c5349a8acd75f323f558fd4fe766/client/my-sites/plan-features/index.jsx#L389-L389?